crates/uv-python/src/discovery.rs

```diff
@@ -2392,4 +2493,89 @@ mod tests {
             )
         );
     }
+
+    #[test]
+    fn executable_names_from_request() {
+        fn case(request: &str, expected: &[&str]) {
+            let (implementation, version) = match PythonRequest::parse(request) {
+                PythonRequest::Any => (None, VersionRequest::Any),
+                PythonRequest::Default => (None, VersionRequest::Default),
+                PythonRequest::Version(version) => (None, version),
+                PythonRequest::ImplementationVersion(implementation, version) => {
+                    (Some(implementation), version)
+                }
+                PythonRequest::Implementation(implementation) => {
+                    (Some(implementation), VersionRequest::Default)
+                }
+                _ => panic!("Test cases should request versions or implemtnations"),
+            };
+
+            let result: Vec<_> = version
+                .executable_names(implementation.as_ref())
+                .into_iter()
+                .map(|name| name.to_string())
+                .collect();
+
+            let expected: Vec<_> = expected
+                .iter()
+                .map(|name| format!("{name}{exe}", exe = std::env::consts::EXE_SUFFIX))
+                .collect();
+
+            assert_eq!(result, expected, "mismatch for case \"{request}\"");
+        }
+
+        case(
+            "any",
+            &[
+                "python", "python3", "cpython", "cpython3", "pypy", "pypy3", "pypy", "pypy3",
+                "graalpy", "graalpy3", "graalpy", "graalpy3", "graalpy", "graalpy3", "graalpy",
+                "graalpy3", "cp", "cp3", "cp", "cp3", "cp", "cp3", "cp", "cp3", "cp", "cp3", "cp",
+                "cp3", "cp", "cp3", "cp", "cp3", "pp", "pp3", "pp", "pp3", "pp", "pp3", "pp",
+                "pp3", "pp", "pp3", "pp", "pp3", "pp", "pp3", "pp", "pp3", "pp", "pp3", "pp",
+                "pp3", "pp", "pp3", "pp", "pp3", "pp", "pp3", "pp", "pp3", "pp", "pp3", "pp",
+                "pp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp",
+                "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp",
+                "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp",
+                "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp",
+                "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp", "gp3", "gp",
+                "gp3", "gp", "gp3", "gp", "gp3",
```

Comment:
Are these intentionally repeated?

Reply:
No that's one of the cases I need to fix quick